model/command_args.go

```diff
@@ -20,7 +20,6 @@ type CommandArgs struct {
 	Command         string               `json:"command"`
 	SiteURL         string               `json:"-"`
 	T               goi18n.TranslateFunc `json:"-"`
-	Session         Session              `json:"-"`
```

Comment:
Are we OK introducing a breaking change in this model for plugins? This model is being exposed through the `ExecuteCommand` [Hook API](https://github.com/mattermost/mattermost-server/blob/bf1d9b0ae9f2e8bcaab251e69eb6bb040584812c/plugin/hooks.go#L90). Should we just take away the functionality for this release and put in a comment saying this is deprecated? and eventually remove it 3-4 releases down the road?

Reply:
Right, custom command won't get this because of the `json:"-"` but plugins will. That's actually kinda a bug, I don't think that was intentional. Certainly not for the translation function.

Unfortunately nobody is actually going to see a comment that says this is deprecated.

We could re-institute as deprecated it no problem. It's just that anyone that happens to use it won't work with plugins that eventually use the execute slash command API.

I will re-estate it for this unlikely case. Filed an issue against 6.0: https://mattermost.atlassian.net/browse/MM-26398